### PR TITLE
#2033 Fix topological graph in case powsybl is used

### LINKED ISSUE #2033: Robustness of topological graph
Segfault can occurs in some cases during manipulation of topological graph, in particular when nodes indexes are not contiguous

### PR DESCRIPTION
**Pull request by lecourtoisflo**
closes #2033

### CODE REVIEW

File: dynawo/sources/Common/DYNGraph.h [hunk 1]
```diff
@@ -90,23 +91,23 @@ class Graph {
...
    */
-  void addVertex(unsigned int indexVertex);
+  void addVertex(unsigned int vertexId);
 
   /**
    * @brief add an edge between two vertices
    *
-   * @param indexVertex1 index of th first vertex
-   * @param indexVertex2 index of the second vertex
+   * @param idVertex1 id of th first vertex
```
Comment (marcochiaramello) [inline dynawo/sources/Common/DYNGraph.h:101]: typo: "the"

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestVoltageLevelInterface.cpp [hunk 2]
```diff
@@ -166,6 +166,201 @@ TEST(DataInterfaceTest, VoltageLevel) {
                           .setHighVoltageLimit(420.)
                           .setLowVoltageLimit(380.)
                           .add();
+//   vlIIDM2.getNodeBreakerView().newBusbarSection()
```
Comment (marcochiaramello) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestVoltageLevelInterface.cpp:169]: why this part is commented?

**Review by marcochiaramello — CHANGES_REQUESTED** [review]

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestVoltageLevelInterface.cpp (hunk 2 above)
Comment (lecourtoisflo) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestVoltageLevelInterface.cpp:169]: Relic from developement. will be removed

File: dynawo/sources/Common/DYNGraph.h (hunk 1 above)
Comment (lecourtoisflo) [reply dynawo/sources/Common/DYNGraph.h:101]: OK

**Review by marcochiaramello — APPROVED** [review]

**Review by gautierbureau — APPROVED** [review]

File: dynawo/sources/Common/DYNGraph.cpp [hunk 3]
```diff
@@ -118,14 +120,15 @@ Graph::findAllPaths(const unsigned int& vertexOrigin, const unsigned int& vertex
 
 bool
 Graph::findAllPaths(const unsigned int& vertexOrigin, const unsigned int& vertexExtremity,
-        PathDescription& currentPath, vector<bool> &encountered, list<PathDescription> &paths, FilteredBoostGraph & filteredGraph,
+        PathDescription& currentPath, boost::unordered_set<unsigned int>& encountered, list<PathDescription> &paths, FilteredBoostGraph & filteredGraph,
```
Comment (rosiereflo) [inline dynawo/sources/Common/DYNGraph.cpp:123]: should be const

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: dynawo/sources/Common/DYNGraph.cpp (hunk 3 above)
Comment (gautierbureau) [reply dynawo/sources/Common/DYNGraph.cpp:123]: i am not sure what should be const with the comment could you do it ?

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**